proxy/src/bin/local_proxy.rs

```diff
@@ -201,6 +204,13 @@ async fn main() -> anyhow::Result<()> {
         },
     ));
 
+    let cancel_rps_limit = Vec::leak(args.cancel_rps_limit.clone());
```

Comment:
I looked it up [from the redis_rps_limit](https://github.com/neondatabase/neon/commit/ad0988f27856f8b80f86f808ad2dd4ec90aadac0#diff-65a00b66914e9bc14b701bf5bbaa958d4dd4805695797f00c2a2372d2d6cdca6R381)

I thought that `GlobalRateLimiter` is expected to handle deallocation during its destruction. Since `.into() `consumes the `cancel_rps_limit` and it goes to the `GlobalRateLimiter::new()` params - so there'll be no leaking. However, I'm relatively new to the Rust memory model and may be missing some nuances. Could you provide more details or clarify if there are any potential issues with this approach?

Reply:
That `Into` implementation will or must make clones because it cannot "consume" the slice even if it knew it came from a `Vec::leak`. To reverse a leak you would have to use `Vec::from_raw`.

Regarding the source you mentioned, observe that `RedisPublisherClient::new` takes a static ref to a slice and then will also create a new `Vec` via `Into::into`. Looking at this I don't understand why we use a leak here. Maybe Conrad was experimenting with `Cow<'static, [RateBucketInfo]>` in `GlobalRateLimiter`.

Reply:
Got it, thanks! Still getting use to the memory model :)